sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableClient.java

```diff
@@ -473,7 +473,14 @@ public Response<TableItem> createTableWithResponse(Duration timeout, Context con
             ResponseFormat.RETURN_NO_CONTENT, null, contextValue),
                 TableItemAccessHelper.createItem(new TableResponseProperties().setTableName(tableName)));
 
-        return callWithOptionalTimeout(callable, THREAD_POOL, timeout, logger);
+        try {
```

Comment:
there's a lot of code duplication now with this PR - can you reduce the duplicate code with a new method or method overload

Reply:
@joshfree Originally the duplicated code was pulled into a helper method, but that's what created the issue with the logging in the first place. @vcolin7 had suggested keeping the overload and having the methods that need different logic be special cases. That's definitely an option but I thought it could make traversing the code logic more confusing by having some methods use an overload and have others not.

Reply:
@joshfree My other option was to use `context` to determine how the error messages should be handled, but it felt awkward to have all calls go through an additional logic path just for a few scenarios where things should be processed differently.

Reply:
you could make an internal Enum as the extra parameter to key off of, so that readers of the code can understand the pivot in behavior

Reply:
I decided to go with `context` since it's something that already exists across our SDKs and devs should be familiar with it